src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -940,6 +955,13 @@ public Authentication authenticate(Authentication authentication) throws Authent
                     } else {
                         lastException = e;
                     }
+                } catch (AuthenticationServiceException e) {
+                    final LDAPConfiguration configuration = getConfigurationFor(delegate.configurationId);
+                    LOGGER.log(Level.WARNING,
+                            String.format("Failed communication with ldap server %s (%s)",
+                                    delegate.configurationId, configuration != null ? configuration.getServer() : "null"),
+                            e);
+                    throw e;
```

Comment:
If you are rethrowing anyway, what is the point of logging now? Presumably some caller `catch`ing the exception should be logging it already, right?

Reply:
Yes, but AIUI the goal is that someone can add a custom logger for `hudson.security.LDAPSecurityRealm` and be confident that all connection errors related to LDAP authentication will be shown there with as much information as possible (i.e. displaying the actual LDAP server that had the error). We could wrap the original exception with a more detailed message, but then the errors would not appear in the custom `hudson.security.LDAPSecurityRealm` log. What do you think?

Reply:
No strong opinion.